zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -26,14 +40,100 @@ pub fn founders_reward(height: Height, network: Network) -> Result<Amount<NonNeg
     }
 }
 
+/// Function `FounderAddressChangeInterval` as specified in [protocol specification §7.8][7.8]
+///
+/// [7.8]: https://zips.z.cash/protocol/canopy.pdf#foundersreward
+pub fn founders_address_change_interval() -> Height {
+    let interval = div_ceil(
+        SLOW_START_SHIFT.0 + PRE_BLOSSOM_HALVING_INTERVAL.0,
+        FOUNDERS_ADDRESS_COUNT,
+    );
+    Height(interval)
+}
+
+/// Get the founders reward t-address for the specified block height as described in [protocol specification §7.8][7.8]
+///
+/// [7.8]: https://zips.z.cash/protocol/canopy.pdf#foundersreward
+pub fn founders_reward_address(height: Height, network: Network) -> Result<Address, Error> {
+    let blossom_height = Blossom
+        .activation_height(network)
+        .expect("blossom activation height should be available");
+
+    if !founders_reward_active(height, network) {
+        panic!("no address returned after canopy");
+    }
+
+    let mut adjusted_height = height;
+    if height >= blossom_height {
+        adjusted_height = Height(
+            blossom_height.0
+                + ((height.0 - blossom_height.0) / (BLOSSOM_POW_TARGET_SPACING_RATIO as u32)),
+        );
+    }
```

Comment:
It feels like we do post-blossom height adjustments in a few different parts of the code.

Is there a way to abstract them into a new function?

Reply:
I created a follow up ticket for this: https://github.com/ZcashFoundation/zebra/issues/1311